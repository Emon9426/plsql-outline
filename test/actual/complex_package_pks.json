{
  "nodes": [
    {
      "type": "PACKAGE_HEADER",
      "name": "complex_package",
      "declarationLine": 2,
      "beginLine": null,
      "exceptionLine": null,
      "endLine": null,
      "level": 1,
      "children": [
        {
          "type": "FUNCTION_DECLARATION",
          "name": "calculate_bonus",
          "declarationLine": 19,
          "beginLine": null,
          "exceptionLine": null,
          "endLine": null,
          "level": 2,
          "children": []
        },
        {
          "type": "PROCEDURE_DECLARATION",
          "name": "process_employees",
          "declarationLine": 22,
          "beginLine": null,
          "exceptionLine": null,
          "endLine": null,
          "level": 2,
          "children": []
        },
        {
          "type": "FUNCTION_DECLARATION",
          "name": "format_name",
          "declarationLine": 25,
          "beginLine": null,
          "exceptionLine": null,
          "endLine": null,
          "level": 2,
          "children": []
        }
      ]
    }
  ],
  "metadata": {
    "sourceFile": "complex_package.pks",
    "totalLines": 14,
    "maxNestingDepth": 2
  }
}