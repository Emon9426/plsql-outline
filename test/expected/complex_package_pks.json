{
  "name": "complex_package",
  "type": "PACKAGE",
  "children": [
    {
      "name": "MAX_VALUE",
      "type": "CONSTANT"
    },
    {
      "name": "t_employee",
      "type": "TYPE"
    },
    {
      "name": "t_employee_tab",
      "type": "TYPE"
    },
    {
      "name": "invalid_value",
      "type": "EXCEPTION"
    },
    {
      "name": "calculate_bonus",
      "type": "FUNCTION"
    },
    {
      "name": "process_employees",
      "type": "PROCEDURE"
    },
    {
      "name": "format_name",
      "type": "FUNCTION"
    }
  ]
}
